AEPMessaging/Sources/LiveActivity/Messaging+LiveActivityPublicAPI.swift

```diff
@@ -25,10 +110,221 @@ public extension Messaging {
+                }
+            }
+
+            for await tokenData in Activity<T>.pushToStartTokenUpdates {
+                let tokenHex = tokenData.map { String(format: "%02x", $0) }.joined()
+                if await liveActivityStore.updatePushToken(for: attributeTypeName, token: tokenHex) {
+                    dispatchPushToStartTokenEvent(attributeTypeName: attributeTypeName, token: tokenHex)
+                } else {
+                    Log.debug(label: MessagingConstants.LOG_TAG, "Duplicate push-to-start token for \(attributeTypeName); skipping event.")
+                }
+            }
         }
+    }
+
+    /// Creates and returns a Task that listens for activity updates.
+    ///
+    /// This task observes the `activityUpdates` asynchronous sequence from `ActivityKit` for the given Live Activity type.
+    /// When a new activity starts, it dispatches a start event and creates child tasks to monitor state transitions
+    /// and push token updates specific to that activity.
+    ///
+    /// - Parameters:
+    ///   - type: The concrete type conforming to `LiveActivityAttributes` used to observe Live Activity lifecycle updates.
+    /// - Returns: A `Task` that runs indefinitely, listening for new Live Activities and setting up listeners
+    ///            for their state changes and push token updates. The task completes only if the underlying sequence ends
+    ///            or the task is explicitly cancelled.
+    private static func createActivityUpdatesTask<T: LiveActivityAttributes>(type _: T.Type) -> Task<Void, Never> {
+        Task {
+            let attributeTypeName = T.attributeTypeName
+
+            // Remove this task from storage when the sequence ends.
+            defer {
+                Task {
+                    await activityUpdateTaskStore.removeTask(for: attributeTypeName)
+                }
+            }
+
+            for await activity in Activity<T>.activityUpdates {
+                // Dispatch Live Activity start tracking event
+                dispatchStartEvent(activity: activity)
+
+                // Use task group to manage state and push token updates concurrently.
+                await withTaskGroup(of: Void.self) { group in
+                    // Listen for state updates.
+                    group.addTask {
+                        for await newState in activity.activityStateUpdates {
+                            if newState == .dismissed || newState == .ended {
+                                dispatchStateUpdateEvent(activity: activity, state: newState)
+                            }
+                        }
+                    }
+
+                    // Listen for push token updates for this activity.
+                    // Live Activities Broadcast via channels do not use this token.
+                    group.addTask {
+                        for await newTokenData in activity.pushTokenUpdates {
```

Comment:
Thank you for getting the ActivityState and updateToken tasks done in this PR ! 🙇🏼